lib/setup-nexus.sh

```diff
@@ -60,8 +60,13 @@ skopeo-sync "${ROOTDIR}/docker"
 
 # Tag SAT image as csm-latest
 sat_image="artifactory.algol60.net/sat-docker/stable/cray-sat"
-sat_version="3.25.0"
+# This value is replaced by release.sh at CSM release distribution build time
+sat_version="@SAT_VERSION@"
+# This csm-latest tag is being phased out, but still used as a default
 skopeo-copy "${sat_image}:${sat_version}" "${sat_image}:csm-latest"
+# Bootstrap sat by writing the sat version to the file used by the sat-podman
+# wrapper script. This is later written by the CSM layer of the CFS config.
+echo "${sat_version}" > /opt/cray/etc/sat/version
```

Comment:
The only question I have here is whether `/opt/cray/etc/sat` exists at the time this is run during a fresh install.  It'd be running on the pit node.  Does `sat` get installed there?  I didn't think so...

Reply:
No, to my knowledge, `sat` does not get installed on the PIT node. It would be smart to `mkdir -p /opt/cray/etc/sat` first so this command will work when the directory does not exist yet. If I fix that, I think it would be harmless in a fresh install scenario. Once the management nodes get booted with their images, they should then have `cray-sat-podman` installed, but I don't believe the CFS configuration is executed against the images before they're booted or against the nodes when they boot until later in the process. Thus, the file `/opt/cray/etc/sat/version` will not exist, and `cray-sat-podman` will default to using the `csm-latest` tag (with no caching issues because it's a fresh install), and the explicit tag will later be set by the CFS configuration. I hope that all made sense, and you can confirm.

Reply:
Will this be executed on the ncn-m001 node (not the PIT) during an upgrade scenario at least? That's what I gathered from [the upgrade documentation](https://github.com/Cray-HPE/docs-csm/blob/release/1.6/upgrade/Stage_0_Prerequisites.md#stage-02---prerequisites). We really only need it to bridge the gap until node personalization of the management NCNs is performed in the management-rollout stage of the IUF.

Reply:
Yes - in an upgrade, it'll run on m001.  And then it'll be lost when m001 is upgraded and brought back into the cluster.  But, as you've mentioned, that's fine because CFS will replace it.

This is exactly what I had in mind: `mkdir -p /opt/cray/etc/sat`

The rest of what you said makes sense to me and jives with my understanding.

Reply:
Thanks for the confirmation. I think I just overlooked making sure the directory exists in this script. I will add that.